### PR TITLE
[Merged by Bors] - feat(Combinatorics/SimpleGraph): define the line graph of a simple graph

### PR DESCRIPTION
**Pull request by b-mehta**
This is pretty bare-bones for now.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by YaelDillies — APPROVED** [review]
maintainer merge

File: Mathlib/Combinatorics/SimpleGraph/LineGraph.lean [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+namespace SimpleGraph
+
+variable {V : Type*} {G : SimpleGraph V}
+
+/--
+The line graph of a simple graph `G` has its vertex set as the edges of `G`, and two vertices of
+the line graph are adjacent if the corresponding edges share a vertex in `G`.
+-/
+@[simps]
```
Comment (urkud) [inline Mathlib/Combinatorics/SimpleGraph/LineGraph.lean:29]: I'm not sure that we always want to `simp` away `(lineGraph G e e').Adj`.

Comment (urkud) [reply Mathlib/Combinatorics/SimpleGraph/LineGraph.lean:29]: Otherwise LGTM. Thanks!
bors d+

Comment (b-mehta) [inline Mathlib/Combinatorics/SimpleGraph/LineGraph.lean:29]:
```suggestion
```

File: Mathlib/Combinatorics/SimpleGraph/LineGraph.lean [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+def lineGraph {V : Type*} (G : SimpleGraph V) : SimpleGraph G.edgeSet where
+  Adj e₁ e₂ := e₁ ≠ e₂ ∧ (e₁ ∩ e₂ : Set V).Nonempty
+  symm e₁ e₂ := by intro h; rwa [ne_comm, Set.inter_comm]
+
+lemma lineGraph_adj_iff_exists {e₁ e₂ : G.edgeSet} :
+    (G.lineGraph).Adj e₁ e₂ ↔ e₁ ≠ e₂ ∧ ∃ v, v ∈ (e₁ : Sym2 V) ∧ v ∈ (e₂ : Sym2 V) := by
+  simp [Set.Nonempty]
```
Comment (b-mehta) [inline Mathlib/Combinatorics/SimpleGraph/LineGraph.lean:36]:
```suggestion
  simp [Set.Nonempty, lineGraph]
```

File: Mathlib/Combinatorics/SimpleGraph/LineGraph.lean [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+@[simp] lemma lineGraph_bot : (⊥ : SimpleGraph V).lineGraph = ⊥ := by aesop
```
Comment (b-mehta) [inline Mathlib/Combinatorics/SimpleGraph/LineGraph.lean:38]:
```suggestion
@[simp] lemma lineGraph_bot : (⊥ : SimpleGraph V).lineGraph = ⊥ := by aesop (add simp lineGraph)
```

**Pull request closed without merging**